### PR TITLE
Refine client config I/O contract handling

### PR DESCRIPTION
**Pull request by loocor**
## Summary
- Consolidates client config file parsing, serialization, persistence, and mutation paths behind `document.rs` and `mutate.rs`.
- Removes the old renderer-based path and updates the client service flow, API handlers, and supporting docs to match the new contract.
- Aligns the client config I/O surface with the refactor context for the `hotfix/client-config-parse-contract` branch.
- No schema or migration changes; this is a backend contract/organization change only.

## Testing
- `git diff --check` passed.
- `cargo test` was started but interrupted before completion.
- Additional Rust validation such as `cargo clippy` was not run in this turn.

## Risks
- The refactor touches multiple client service modules, so remaining risk is concentrated in config parse/merge edge cases and mutation behavior.
- Follow-up validation should cover the affected client config flows once the interrupted Rust test run is completed.

### CODE REVIEW

File: backend/src/api/handlers/client/handlers.rs [hunk 1]
```diff
@@ -602,13 +585,29 @@ async fn apply_client_config_request(
...
+                    error = %err,
+                    "Invalid persisted config_file_parse while building apply preview"
+                );
+                StatusCode::INTERNAL_SERVER_ERROR
+            })?;
+            parse_config_fallback(
+                Some(preview_content),
+                parse_rule.as_ref(),
+                state.config_path(),
+            )
```
[the remark this answers is not in the document]
Comment (loocor) [reply backend/src/api/handlers/client/handlers.rs]: Fixed in 63f8585. Apply handler now degrades gracefully — when `effective_config_file_parse` fails, it logs a warning and falls back to `parse_config_fallback` with autodetect instead of returning 500. The config is already persisted at that point, so the preview is purely informational.

File: backend/src/api/handlers/client/handlers.rs [hunk 2]
```diff
@@ -300,8 +274,17 @@ pub async fn config_details(
...
+            Ok(parse_rule) => parse_config_fallback(
+                content.as_deref(),
+                parse_rule.as_ref(),
+                state.config_path(),
+            ),
+            Err(err) => {
+                warnings.push(format!("Persisted config parse metadata is invalid: {err}"));
+                degraded_reasons.push("config_file_parse_invalid".to_string());
+                Value::Null
+            }
```
[the remark this answers is not in the document]
Comment (loocor) [reply backend/src/api/handlers/client/handlers.rs:287]: Fixed in 63f8585. Config details handler now falls back to `parse_config_fallback(content, None, config_path)` when parse metadata is invalid — this tries path-hinted autodetect and returns raw string as last resort, instead of returning `Value::Null`. Warning and degraded_reason are still emitted.
